crates/full-node/sov-sequencer/src/preferred/mod.rs

```diff
@@ -130,6 +199,24 @@ where
         stop_at_rollup_height: Option<RollupHeight>,
     ) -> anyhow::Result<(Arc<Self>, Vec<JoinHandle<()>>)> {
         let shutdown_receiver = shutdown_sender.subscribe();
+        let mut config = config.clone();
+        let oracle_key = match &config.sequencer_kind_config.timing_oracle.private_key_hex {
```

Comment:
Let's extract all the oracle code to a separate file and only call it in the preffered/mod.rs